src/main/perl/Element.pm

```diff
@@ -559,33 +585,62 @@ sub getTree
 
 SWITCH:
     {
+        # LIST to array ref
         $self->isType(LIST) && do {
             $ret = [];
             while ($self->hasNextElement) {
                 $el = $self->getNextElement();
-                push(@$ret, $el->getTree($nextdepth));
+                push(@$ret, $el->getTree($nextdepth, %opts));
             }
             last SWITCH;
         };
+
+        # NLIST to hashref
         $self->isType(NLIST) && do {
             $ret = {};
             while ($self->hasNextElement) {
                 $el = $self->getNextElement();
-                $$ret{$el->getName()} = $el->getTree($nextdepth);
+                $$ret{$el->getName()} = $el->getTree($nextdepth, %opts);
             }
             last SWITCH;
         };
+
+        # BOOLEAN via convert_bool method
         $self->isType(BOOLEAN) && do {
             $ret = $self->getValue eq 'true' ? 1 : 0;
+            $convm = $opts{convert_boolean};
+            last SWITCH;
+        };
+
+        # STRING to string
+        $self->isType(STRING) && do {
+            $ret = $self->getValue;
+            $ret = "$ret";
+            $convm = $opts{convert_string};
+            last SWITCH;
+        };
+
+        # LONG to integer/long
+        $self->isType(LONG) && do {
+            $ret = 0 + $self->getValue;
+            $convm = $opts{convert_long};
+            last SWITCH;
+        };
+
+        # DOUBLE to float/double
+        $self->isType(DOUBLE) && do {
+            $ret = 0.0 + $self->getValue;
+            $convm = $opts{convert_double};
             last SWITCH;
         };
 
-        # Default clause
+        # Default clause, should never be reached
```

Comment:
because there's no logger here, only `throw_error`. i don't mind throwing it, but others might disagree
@ned21 toughts?